### DISCUSSION TITLE
Spreading stylex.props in a component vs an html element

### DISCUSSION THREAD

**Discussion by sumanbh** [facebook/stylex#766] [General]
In context of headless components where we're trying to apply the initial style, say we have [DialogPanel](https://headlessui.com/react/dialog) from headlessui, is it recommended/preferred to do this

```tsx
<DialogPanel {...stylex.props(...)}>
  ...
</DialogPanel>
```

vs

```tsx
<DialogPanel as={Fragment}>
  <div {...stylex.props(...)}>...</div>
</DialogPanel>
```

**Comment by nmn** [comment]
Yes, if a custom component, such as `DialogPanel` accepts both `className` and `style` (and `data-` attributes), then `<DialogPanel {...stylex.props(...)}>` is preferred.

---

A component library built for StyleX should accept a `style` prop that accepts StyleX styles instead and use `stylex.props` internally.
